SIPS/sip-11.md

```diff
@@ -0,0 +1,76 @@
+---
+sip: 11
+title: Transaction Insights V2 - Transaction Warnings
+status: Draft
+discussions-to: https://github.com/MetaMask/SIPs/discussions/101
+author: Hassan Malik (@hmalik88)
+created: 2023-07-16
+---
+
+## Abstract
+
+This SIP proposes a way for snaps to provide extra friction for executing potentially dangerous transactions within MetaMask.
+
+## Motivation
+
+One of the biggest issues with wallet users is a loss of funds through executing some sketchy or seemingly innocuous transaction. With the benefit of the transaction insights API, a snap can already run analysis on an unsigned transaction paylod. Why not allow for the snap to provide extra friction to the user if it determines that a transaction is malicious? Users often click through things without reading, by allowing for a warning we can add friction at points that we really think a user should have a second glance.
+
+This warning could be injected at any point where there are insights provided.
+
+## Specification
+
+Please see [SIP-3](https://github.com/MetaMask/SIPs/blob/main/SIPS/sip-3.md) for more information on the original transaction insights API.
+
+Please see the [@metamask/snaps-ui](https://github.com/MetaMask/snaps/blob/main/packages/snaps-ui/src/nodes.ts) package for more information on the `Component` type returned in the `OnTransactionResponse`.
```

Comment:
This should link to the UI SIP.

Reply:
I was doing that at first, but it's outdated the return type outlined there is not the most up-to-date one.